src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressByMessageFilterTest.java

```diff
@@ -0,0 +1,164 @@
+        config.addProperty("messageFormat", "('[a-z]{1}')");
+
+        final String[] expectedViolations = {
+            "25:9: " + getCheckMessage(RequireThisCheck.class, MSG_METHOD, "foo", ""),
+        };
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter5.java"),
+                expectedViolations);
+    }
+
+    @Test
+    public void testMatchMessageMethod() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+        config.addProperty("messageFormat", "('foo')");
+
+        final String[] expectedViolations = {
+            "23:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "a", ""),
+            "24:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "b", ""),
+            "29:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "c", ""),
+        };
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter6.java"),
+                expectedViolations);
+    }
+
+    @Test
+    public void testNoMatchMessage() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+        config.addProperty("messageFormat", "Nonsense1234Regex");
+
+        final String[] expectedViolations = {
+            "23:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "a", ""),
+            "24:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "b", ""),
+            "25:9: " + getCheckMessage(RequireThisCheck.class, MSG_METHOD, "foo", ""),
+            "29:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "c", ""),
+        };
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter7.java"),
+                expectedViolations);
+    }
+
+    @Test
+    public void testMatchCheckAndMessage() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+        config.addProperty("checkFormat", "RequireThisCheck");
+        config.addProperty("messageFormat", "needs");
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter8.java"),
+                CommonUtil.EMPTY_STRING_ARRAY);
+    }
+
+    private void verifySuppressed(DefaultConfiguration config, String fileName,
+                                  String... expectedViolations) throws Exception {
+        final DefaultConfiguration requireThisConfig =
+                createModuleConfig(RequireThisCheck.class);
+        requireThisConfig.addProperty("validateOnlyOverlapping", "false");
+
+        final DefaultConfiguration treewalkerConfig = createModuleConfig(TreeWalker.class);
+        treewalkerConfig.addChild(requireThisConfig);
+        final DefaultConfiguration checkerConfig = createRootConfig(treewalkerConfig);
+        checkerConfig.addChild(config);
+
+        verify(checkerConfig, fileName, expectedViolations);
```

Comment:
This is a forbidden method. Please use parser.

Reply:
I used parser and now I am getting this error:
`com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module com.puppycrawl.tools.checkstyle.TreeWalker - TreeWalker is not allowed as a parent of com.puppycrawl.tools.checkstyle.filters.SuppressByMessageFilter Please review 'Parent Module' section for this Check in web documentation if Check is standard.`

Reply:
I took a reference from https://github.com/checkstyle/checkstyle/pull/10955
Should I have to use verifyFilterWithInlineConfigParser?

Reply:
> Should I have to use verifyFilterWithInlineConfigParser?

yes

Reply:
Can you tell me what is suppressed violation?

Reply:
Do I need to remove the `verifySuppressed` method and add `verifyFilterWithInlineConfigParser` in the UTs or I have to 
just use the parser in the `verifySuppressed` Method?

Reply:
You can do either of them.

> just use the parser in the verifySuppressed Method?

I'll suggest this to avoid repeatition.

Reply:
But in that case, I have to make `unfiltered violations` or how can I use my `checkerConfig` in the parser?

Reply:
when I make it like:
`verifyFilterWithInlineConfigParser(fileName, expectedViolations);`
I find these issue: https://github.com/checkstyle/checkstyle/pull/11174#discussion_r787415560

Reply:
@romani please help!

Reply:
@rnveach please help!

Reply:
@nmancus1  need your help.

Reply:
@romani  sir, please have a look at this and let me know how can I resolve this? https://github.com/checkstyle/checkstyle/pull/11174#discussion_r788918331

Reply:
Have you tried looking at how other tests used inline config parser?
2e46a394a0c2f86b15d3951fed17c50c814b38c3

Reply:
@rnveach @romani I have seen and understood that I need to add the `removeSuppressed` method but can you please tell me more about the removeSuppressed Method so that I can add this to my code.